pkg/internal/resources/fakeoperator/fakeoperator.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fakeoperator
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/kustomize/v3/pkg/image"
+	"sigs.k8s.io/kustomize/v3/pkg/types"
+
+	"github.com/kubermatic/kubecarrier/pkg/internal/kustomize"
+	"github.com/kubermatic/kubecarrier/pkg/internal/version"
+)
+
+// Config holds the config information to generate the kubecarrier operator setup.
+type Config struct {
+	// Namespace the e2e operator should be deployed into.
+	Namespace string
+	// Name of this e2e operator object
+	Name string
+}
+
+var k = kustomize.NewDefaultKustomize()
+
+func Manifests(c Config) ([]unstructured.Unstructured, error) {
+	v := version.Get()
+	kc := k.ForHTTP(vfs)
+	if err := kc.MkLayer("man", types.Kustomization{
```

Comment:
Maybe add `NamePrefix: c.Name + "-"` as well.

Reply:
do I really need it if I already have `namePrefix` in `kustomization.yaml`?

Reply:
It's a good practice (in this case not really since we won't spin two e2e operators, but let's keep things consistent.

Reply:
I thought we decided to get rid of `Name` because we don't use it anywhere else, except test. We consistent with `manager`, when we have only one instance, we don't use namePrefix